### PR TITLE
52 iterator refactor

### PR DESCRIPTION
**Pull request by peiva-git**

### CODE REVIEW

File: src/main/java/FreedomPointsCounter.java [hunk 1]
```diff
@@ -50,7 +49,8 @@ private void checkAllFreedomLinesFrom(@NotNull Position position) {
     }
 
     private void checkFreedomLine(@NotNull Position startingPosition, @NotNull Direction direction) {
-        Stone.Color stoneColor = board.getStone(startingPosition).getColor();
+        Stone startingStone = Objects.requireNonNull(board.getStone(startingPosition), "Should be not-null, checked by count method");
```
Comment (peiva-git) [inline src/main/java/FreedomPointsCounter.java:52]: It's not entirely clear to me whether the other cases need to be checked for `null` values. Please take a look yourself as well

Comment (manuelkoso) [reply src/main/java/FreedomPointsCounter.java:52]: Ok, I will refactor this thing and I will add some documentation

**Review by manuelkoso — APPROVED** [review]
It seems to me ok

**Comment by manuelkoso** [comment]
I will open a new pull request for the `FreedomPointsCounter` class. This pull request is regarding the iterator. Is it ok for you?

**Comment by peiva-git** [comment]
Ok, we can keep the issue then, #54 will close only #52

**Pull request merged**